packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -413,9 +418,11 @@ class FlutterSkippingFileComparator extends FlutterGoldenFileComparator {
   /// If we are in a CI environment, LUCI or Cirrus, but are not using the other
   /// comparators, we skip.
   static bool isAvailableForEnvironment(Platform platform) {
-    return platform.environment.containsKey('SWARMING_TASK_ID')
+    return (platform.environment.containsKey('SWARMING_TASK_ID')
       // Some builds are still being run on Cirrus, we should skip these.
-      || platform.environment.containsKey('CIRRUS_CI');
+      || platform.environment.containsKey('CIRRUS_CI'))
+      // If we are in CI, skip on branches that are not main.
+      && !_kMainBranch.hasMatch(platform.environment['GIT_BRANCH'] ?? '');
```

Comment:
this doesn't seem right? it makes us use the local file comparator on the main branch on Cirrus, for example, whereas before we used the skipping one.

Reply:
(i'll fix it in my PR, let me know if i misunderstood this)

Reply:
Oh. Well gosh. Yeah that's wrong, and obviously not covered by a test.
We don't actually run these tests on Cirrus at all anymore. I had the thought to clean everything Cirrus related out while I was here, but didn't want to make merging with your PR even worse.